pilot/pkg/model/virtualservice.go

```diff
@@ -364,8 +382,30 @@ func hasConflict(root, leaf *networking.HTTPMatchRequest) bool {
 		return true
 	}
 
-	// sourceLabels, sourceNamespace and gateways do not apply to delegate
-	// and are empty, so no conflict for them.
+	// sourceNamespace
+	if root.SourceNamespace != "" && leaf.SourceNamespace != root.SourceNamespace {
+		return true
+	}
+
+	// sourceLabels
```

Comment:
Can you please add more comments here and below
- Labels should not be conflicting?
- Root should have super set of gateways?